Sources/Controllers/2.0/RoutePlanning/OASaveTrackViewController.mm

```diff
@@ -528,6 +576,7 @@ - (void) onItemSelected:(int)index
 {
     _selectedFolderIndex = index;
     _selectedFolderName = _allFolders[index];
+    [self updateErrorMessage:_fileName];
     [self generateData];
     [self.tableView reloadData];
```

Comment:
Do not reload the entire tableView if only one or two cells were changed. Use reload rows/sections methods instead. Do it not only here but for the entire class. (And possibly for other classes where you see similar code)